### PR TITLE
fix: global agent infinite connections

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
@dnlup @Ethan-Arrowood

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/pool.js [hunk 1]
```diff
@@ -14,28 +14,33 @@ const kQueue = Symbol('queue')
...
 
-    if (connections != null && (!Number.isFinite(connections) || connections <= 0)) {
+    const { connections } = options
+
+    if (connections != null && (!Number.isFinite(connections) || connections < 0)) {
       throw new InvalidArgumentError('invalid connections')
     }
 
+    this[kUrl] = url
+    this[kOptions] = JSON.parse(JSON.stringify(options))
```
Comment (Ethan-Arrowood) [inline lib/pool.js:34]: What scenario is this covering? Failing if `options` is not an object?

**Review by dnlup — APPROVED** [review]
lgtm

File: lib/pool.js (hunk 1 above)
Comment (ronag) [reply lib/pool.js:34]: it's to guard against mutations

**Review by Ethan-Arrowood — APPROVED** [review]

File: lib/pool.js (hunk 1 above)
Comment (mcollina) [reply lib/pool.js:34]: This is also a known performance issue

**Pull request merged**